Ark.Tools.Hosting/LegacyEnvironmentVariablesConfigurationProvider.cs

```diff
@@ -0,0 +1,145 @@
+using System.Linq;
+using System.Runtime;
+
+namespace Microsoft.Extensions.Configuration.EnvironmentVariables
+{
+    /// <summary>
+    /// An environment variable based <see cref="ConfigurationProvider"/>.
+    /// </summary>
+    public class ArkLegacyEnvironmentVariablesConfigurationProvider : ConfigurationProvider
+    {
+        private const string _mySqlServerPrefix = "MYSQLCONNSTR_";
+        private const string _sqlAzureServerPrefix = "SQLAZURECONNSTR_";
+        private const string _sqlServerPrefix = "SQLCONNSTR_";
+        private const string _customPrefix = "CUSTOMCONNSTR_";
+
+        private const string _connStrKeyFormat = _connStrKey+"{0}";
+        private const string _connStrKey = "ConnectionStrings:";
+        private const string _providerKeyFormat = "ConnectionStrings:{0}_ProviderName";
+
+        private readonly string _prefix;
+
+        /// <summary>
+        /// Initializes a new instance.
+        /// </summary>
+        public ArkLegacyEnvironmentVariablesConfigurationProvider() : this(string.Empty)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance with the specified prefix.
+        /// </summary>
+        /// <param name="prefix">A prefix used to filter the environment variables.</param>
+        public ArkLegacyEnvironmentVariablesConfigurationProvider(string prefix)
+        {
+            _prefix = prefix ?? string.Empty;
+        }
+
+        /// <summary>
+        /// Loads the environment variables.
+        /// </summary>
+        public override void Load()
+        {
+            Load(Environment.GetEnvironmentVariables());
+        }
+
+        internal void Load(IDictionary envVariables)
+        {
+            var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            var filteredEnvVariables = envVariables
+                .Cast<DictionaryEntry>()
+                .SelectMany(_azureEnvToAppEnv);
+
+            filteredEnvVariables = _duplicateConnectionStringForDifferentEnvinroment(filteredEnvVariables)
+                .Where(entry => ((string)entry.Key).StartsWith(_prefix, StringComparison.OrdinalIgnoreCase));
+
+            foreach (var envVariable in filteredEnvVariables)
+            {
+                var key = ((string)envVariable.Key).Substring(_prefix.Length);
+                data[key] = envVariable.Value != null ? (string)envVariable.Value : String.Empty;
+            }
+
+            Data = data;
+        }
+
+        private static string _normalizeConnectionStringKey(string key)
+        {
+            return key.Replace("__", ConfigurationPath.KeyDelimiter);
+        }
+
+        private static string _normalizeAppSettingsKey(string key)
+        {
+            return _normalizeConnectionStringKey(key).Replace(".", ConfigurationPath.KeyDelimiter);
+        }
+
+        private IEnumerable<DictionaryEntry> _duplicateConnectionStringForDifferentEnvinroment(IEnumerable<DictionaryEntry> filteredEnvVariables)
```

Comment:
thinking about this, is probably not needed as both this 'LegacyEnvironmentVariable' **and** the library native 'EnvironementVariable' providers can be registered.

try remove the duplication and test this setup:
1. Host.CreateHostDefaults(....)
2. ConfigureAppConfiguration((ctx, cfg) =>cfg.AddArkLegacyEnvironmentVariables())
3. Add in lunch.json the env variable CUSTOMCONNSTR_WORKER_DATABASE=whatever
4. try Configuration.GetConnectionString("Worker.Database") and Configuration.GetConnectionString("Worker_Database") and both should work fine.